_posts/2016-05-30-one-model-to-rule-them-all.md

```diff
@@ -0,0 +1,154 @@
+
+Fortunately, our model handles these discrepancies by extending the base class:
+
+{% highlight javascript %}
+class Android extends Login { ... }
+
+// No ID on the Android app :(. No problem, XPath for the rescue!
+Android.EMAIL_INPUT = '//android.widget.EditText';
+
+class IOS extends Login {
+  setEmail(email) {
+    // On iOS we can directly inject the value, and it's way faster!
+    this.driver.element(this.constructor.EMAIL_INPUT).setValue(email);
+    return this;
+  }
+}
+{% endhighlight %}
+
+Great! Now we can take into account platform specificities while having most of our code shared. Now, what if we want to reuse our login form model for a signup form? We had to rethink the Page Object pattern and come up with a new paradigm. Using a mix-in-like construction, we can inject new methods from the `SignUp` class into parent classes to create new classes which benefit from previously-implemented behaviors and locators. `SignUp` works as an interface with default methods:
+
+{% highlight javascript %}
+const SignUp = Page => class extends Page {
+  setUserName(name) { ... }
+  setBirthdate(date) { ... }
+}
+// `login` is an object which has an IOS and an Android class as members
+class Android extends SignUp(login.Android) { ... }
+class IOS extends SignUp(login.IOS) { ... }
+{% endhighlight %}
+
+We call this construction a *Neutron* in Qualcium. It is a component comprised of one interface and two classes, each one representing a platform, just as [the subatomic particle](https://en.wikipedia.org/wiki/Neutron) consists of one up quark and two down quarks. Neutrons are modules, and they adopt a common standard:
+
+<img src="/images/2016-05-30-one-model-to-rule-them-all/neutrons.png" class="block" style="width: 352px;" />
+
+Using this standard allows us to create reusable Neutrons from other Neutrons while also addressing our code maintainability requirement.
+
+## Fluently describe flows
+
+One of our goals when designing such models is to have them provide sets of methods that represent high-level actions, avoiding or hiding less useful low-level actions as much as possible. Additionally, we want Neutrons' interfaces to abstract away divergent behavior, describing what the service offered by the application does, not how it works. Last but not least, we want the relationships between these Neutrons to **reflect the user's journey through the application**.
```

Comment:
Can you please italicize "what" and "how" in "[...] describing _what_ the service offered by the application does, not _how_ it works." ?